src/shogun/converter/ica/ICAConverter.h

```diff
@@ -30,10 +30,22 @@ class CICAConverter: public CConverter
 		/** destructor */
 		virtual ~CICAConverter();
 
-		/** apply to features
-		 * @param features features to embed
+		/** Fit the ICA converter to features
+		 * Subclasses should implement this method to calculate the mixing
+		 * matrix
+		 * @param features the training features, should be an instance of
+		 * CDenseFeatures<float64_t>
 		 */
-		virtual CFeatures* apply(CFeatures* features) = 0;
+		virtual void fit(CFeatures* features) = 0;
```

Comment:
if we know that this class is an abstract class for other classes that does conversion over `CDenseFeatures<float64_t>` then here it would be good to actually do the type mapping, meaning not to have this function pure virtual but implement something like this:

```
virtual void fit(CFeatures* features)
{
    REQUIRE(features->get_feature_class() == C_DENSE, "ICA converters only work with dense features")
    fit((CDenseFeatures<float64_t>*)features);
}
```

and have a pure virtual protected member function that has a declaration:

```
virtual void fit(CDenseFeatures<float64_t>*) = 0;
```

this way the casting thing you need to do it in one place, all the other places you can already assume CDenseFeatures<float64_t> as input.